server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -176,14 +170,36 @@ void updateSecretSeries(
   }
 
   public int setExpiration(long secretContentId, Instant expiration) {
-    Field<Long> minExpiration = decode()
-        .when(SECRETS_CONTENT.EXPIRY.eq(0L), val(expiration.getEpochSecond()))
-        .otherwise(least(SECRETS_CONTENT.EXPIRY, val(expiration.getEpochSecond())));
```

Comment:
decode() appears to be a method which allows you to insert conditional logic as a value in a Jooq query. As a consequence of the edits below (reading the record first rather than doing a blind update), this has been replaced with plain Java.